homepage/views.py

```diff
@@ -77,3 +80,34 @@ def add_review_search(request):
             course_name = ''
         courses = Course.get_courses_ordered_by_name(course_name)
         return render(request, 'homepage/add_review_search.html', {'course_name': course_name, 'courses': courses})
+
+
+def sign_in(request):
+    signin_successful = False
+
+    if request.method == "POST":
+        form = AuthenticationForm(request, data=request.POST)
+
+        if form.is_valid():
+            username = form.cleaned_data.get('username')
+            password = form.cleaned_data.get('password')
+            user = authenticate(username=username, password=password)
+
+            if user is not None:
```

Comment:
maybe you can do it like:
```suggestion
            if user:
```

Reply:
changed